### PR TITLE
Revert to wait for frame for R200, F200, SR300 cameras

### PR DESCRIPTION
**Pull request by mkhansenbot**
**Please ensure the above *guidelines for contributing* are met.**

Fixes Issue: #210, #207

Changes proposed in this pull request:
- Adds new synchronous base class which uses rs_wait_for_frames for use with the F200, SR300, and R200 cameras; this is to improve the reliability when running RGBD, which relies on synchronous color & depth streams

### CODE REVIEW

File: realsense_camera/src/sync_nodelet.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+   */
+  void SyncNodelet::onInit()
+  {
+    BaseNodelet::onInit();
+
+    // start thread to publish topics
+    topic_thread_ =
+            boost::shared_ptr <boost::thread>(new boost::thread (boost::bind(&SyncNodelet::publishSyncTopics, this)));
+  }
+
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp:53]: Please add a destructor for this class which does the thread join for the topic_thread_

topic_thread_->join();

File: realsense_camera/src/sync_nodelet.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      {
+        if (start_camera_ == true)
+        {
+          ROS_INFO_STREAM(nodelet_name_ << " - " << startCamera());
+        }
+        else
+        {
+          ROS_INFO_STREAM(nodelet_name_ << " - " << stopCamera());
+        }
+        start_stop_srv_called_ = false;
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp]: please add a blank line

File: realsense_camera/src/sync_nodelet.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+        if (pointcloud_publisher_.getNumSubscribers() > 0 &&
+            rs_is_stream_enabled(rs_device_, RS_STREAM_DEPTH, 0) == 1 && enable_pointcloud_ == true &&
+            (duplicate_depth_color_ == false))  // Skip publishing PointCloud if Depth or Color frame was duplicate
+        {
+          if (camera_publisher_[RS_STREAM_DEPTH].getNumSubscribers() <= 0)
+          {
+            setImageData(RS_STREAM_DEPTH);
+          }
+          if (camera_publisher_[RS_STREAM_COLOR].getNumSubscribers() <= 0)
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp]: please add a blank line

File: realsense_camera/src/sync_nodelet.cpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          {
+            setImageData(RS_STREAM_COLOR);
+          }
+          publishPCTopic();
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp]: please add a blank line

File: realsense_camera/src/sync_nodelet.cpp [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        camera_publisher_[stream_index].publish(msg, camera_info_ptr_[stream_index]);
+      }
+      else
+      {
+        if ((stream_index == RS_STREAM_DEPTH) || (stream_index == RS_STREAM_COLOR))
+        {
+          duplicate_depth_color_ = true;  // Set this flag to true if Depth and/or Color frame is duplicate
+        }
+      }
+      ts_[stream_index] = frame_ts;
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp]: please add a blank line

File: realsense_camera/src/sync_nodelet.cpp [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    }
+  }
+  void SyncNodelet::setImageData(rs_stream stream_index)
```
Comment (mdhorn) [inline realsense_camera/src/sync_nodelet.cpp]: please add a blank line

**Review by mdhorn — CHANGES_REQUESTED** [review]

**Review by mdhorn — APPROVED** [review]
LGTM 👍

**Review by mattcurfman — COMMENTED** [review]
Looks good, thank you.

**Review by mattcurfman — APPROVED** [review]

**Pull request merged**